File: codeforces/problemset/B. Pleasant Pairs.cpp
#include <bits/stdc++.h>
using namespace std;
void solve(){
    int n, cnt = 0;
    scanf("%d", &n);
    vector<pair<long long, long long>>a(n);
    for(int i = 0;i < n;i++){
        scanf("%lld", &a[i].first);
        a[i].second = i + 1;
    }
    sort(a.begin(),a.end());
    for(int i = 0;i < n;i++){
        for(int j = i + 1;j < n;j++){
            if(a[i].first * a[j].first > 2 * n)
                break;
            if(a[i].first * a[j].first == a[i].second + a[j].second)
                cnt++;
        }
    }
    cout << cnt << '\n';

}
int main(){ 

    int t;
    scanf("%d",&t);
    while(t--){
        solve();
    }

    return 0;
}

File: codeforces/problemset/B. Pleasant Pairs.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class Pair {
    int first;
    int second;

    public Pair(int first, int second) {
        this.first = first;
        this.second = second;
    }
}

public class Main {

    public static void solve(Scanner scanner) {
        int n = scanner.nextInt();
        List<Pair> a = new ArrayList<>();

        // Reading the pairs
        for (int i = 0; i < n; i++) {
            int first = scanner.nextInt();
            a.add(new Pair(first, 0)); // Initially set the second value to 0 or any default value
        }

        // You can now add more processing logic here as needed
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        
        while (t-- > 0) {
            solve(scanner);
        }

        scanner.close();
    }
}
